src/NaturalEarth.jl

```diff
@@ -50,5 +65,195 @@ function bathymetry(contour::Int=2000)
 end
 
 
+# Get rasters from AWS
+
+# function get_available_rasters(refresh = false)
+#     scratchspace = Scratch.@get_scratch!("rasters")
+#     xml_file = joinpath(scratchspace, "index.xml")
+#     # refresh the XML file
+#     if !isfile(xml_file) || refresh
+#         download("https://naturalearth.s3.amazonaws.com/", xml_file)
+#     end
+#     # read the document
+#     xml_doc = LightXML.parse_file(xml_file);
+#     # get the root node
+#     xml_root = LightXML.root(xml_doc);
+#     @assert name(xml_root) == "ListBucketResult" "The format of the NaturalEarth data dictionary at Amazon AWS has changed, or the file is invalid.  Please file an issue at https://github.com/JuliaGeo/NaturalEarth.jl/issues."
+#     return deepcopy(attributes_dict(xml_root))
+# end
+
+# function available_rasters_dict(keys)
+#     retval = Dict{String, Vector{VersionNumber}}()
+#     for key_element in keys
+#         key_str = key_element[1]
+#         endswith(key_str, "/") && continue # discard directory specs
+#         if haskey(key_str, "raster")
+#             name = splitext(basename(key_str))[1]
+#             dirs = splitpath(name)
+
+#             version = if length(dirs) == 1 || !(occursin(".", dirs[1])) # no version number
+#                 VersionNumber(typemax(UInt32))
+#             else
+#                 VersionNumber(dirs[1])
+#             end
+
+#             # for i in 1
+#     end
+# end
+
+# Implement the RasterDataSources.jl interface.
+# With this, you can call e.g. # Rasters.Raster(NaturalEarthRaster{:latest, 10, :HYP_HR}, nothing)
+
+struct NaturalEarthRaster{Version, Scale, Name} <: RasterDataSources.RasterDataSource end
```

Comment:
The way RasterDataSources works is that you have to pass a type in to `RasterDataSources.getraster`, and Rasters.jl has the same dispatch (`Raster(T::Type{<: RasterDataSources.RasterDataSource}, layer; kwargs...)`).

Reply:
I do wish we could keep them as fields, since that would allow a lot more validation, but oh well 🤷

Reply:
I would put less in the type parameter!

RasterDataSources.jl has tge layer name as the second argument, and scale would be a keyword. 

Otherwise you get compilation every time you use it. Having the name as second argument mesns a tuple of names can make a stack.

Reply:
To clarify, when there are multiple type parameters in RasterDataSources its usually because the data source is structurally different - they have different keyword arguments and separate dataset handling code. `WorldClim{Weather}` and `WorldClim{Climate}` dispatch to totally separate methods in different files.

Here you don't actually use them to dispatch different code, suggesting they probably should be keywords.

There are also a few in RasterDataSources.jl liuke `ALWB` that were just design mistakes and have unnecessary things in the type that could be keywords.